src/app/services/technical-record/technical-record.service.ts

```diff
@@ -416,6 +422,57 @@ export class TechnicalRecordService {
     };
   }
 
+  validateForBatch(): AsyncValidatorFn {
+    return (control: AbstractControl): Observable<ValidationErrors | null> => {
+      const trailerIdControl = control.parent?.get('trailerId') as CustomFormControl;
+      const vinControl = control.parent?.get('vin') as CustomFormControl;
+      const systemNumberControl = control.parent?.get('systemNumber') as CustomFormControl;
+      if (trailerIdControl && vinControl) {
+        const trailerId = trailerIdControl.value;
+        const vin = vinControl.value;
+        delete vinControl.meta.warning;
+
+        if (trailerId && vin) {
+          return this.getByVin(vin).pipe(
+            map(result => {
+              const filteredResults = result.filter(vehicleTechRecord => vehicleTechRecord.trailerId === trailerId);
+              if (!result || !filteredResults.length) {
```

Comment:
I'm confused by this. You're calling `.filter` on something that could be undefined according to line 439. So it's one or the other.